src/main/kotlin/org/misarch/catalog/graphql/model/ProductVariant.kt

```diff
@@ -0,0 +1,60 @@
+package org.misarch.catalog.graphql.model
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLIgnore
+import kotlinx.coroutines.reactor.awaitSingle
+import org.misarch.catalog.graphql.model.connection.ProductVariantVersionConnection
+import org.misarch.catalog.graphql.model.connection.ProductVariantVersionOrder
+import org.misarch.catalog.persistance.model.ProductVariantVersionEntity
+import org.misarch.catalog.persistance.repository.ProductRepository
+import org.misarch.catalog.persistance.repository.ProductVariantVersionRepository
+import org.springframework.beans.factory.annotation.Autowired
+import java.util.*
+
+@GraphQLDescription("A variant of a Product.")
+class ProductVariant(
+    id: UUID,
+    @GraphQLDescription("If true, the ProductVariant is visible to customers.")
+    val isPubliclyVisible: Boolean, private val productId: UUID, private val currentVersion: UUID
+) : Node(id) {
+
+    @GraphQLDescription("The Product this is a ProductVariant of.")
+    suspend fun product(
+        @Autowired
+        @GraphQLIgnore
+        productRepository: ProductRepository
+    ): Product {
+        return productRepository.findById(productId).awaitSingle().toDTO()
+    }
+
+    @GraphQLDescription("The current version of the ProductVariant.")
+    suspend fun currentVersion(
+        @Autowired
+        @GraphQLIgnore
+        productVariantVersionRepository: ProductVariantVersionRepository
+    ): ProductVariantVersion {
+        return productVariantVersionRepository.findById(currentVersion).awaitSingle().toDTO()
+    }
+
+    @GraphQLDescription("Get all associated versions")
+    suspend fun versions(
+        @GraphQLDescription("Number of items to return")
+        first: Int? = null,
+        @GraphQLDescription("Number of items to skip")
+        skip: Int? = null,
+        @GraphQLDescription("Ordering")
+        orderBy: ProductVariantVersionOrder? = null,
```

Comment:
Isn't the default here `DESC`?

Reply:
imho consistency for all connections is more important, but I think this is up to discussion

Reply:
Well… Yes, of course, I like consistency too.
However, in this case, you have the inherent constraint  `show the most recent version on top`, so I don't think it makes sense not to require a default sorting here.

Reply:
imho if you want the current version, you can just use that field
tbh we could discuss if we even need to support all versions, if so I think we should limit the access rights here to employee or so